src/store/hotspotDetails/hotspotDetailsSlice.ts

```diff
@@ -102,9 +109,22 @@ export const fetchHotspotChallengeSums = async (
   }
 }
 
-export const fetchHotspotDetails = createAsyncThunk(
+export const fetchHotspotDetails = createAsyncThunk<
+  HotspotDetailsObj,
+  FetchDetailsParams
+>(
   'hotspotDetails/fetchHotspotDetails',
-  async (params: FetchDetailsParams) => {
+  async (params: FetchDetailsParams, { getState }) => {
+    const currentState = (getState() as {
+      hotspotDetails: {
+        details: Record<string, Record<number, HotspotDetails>>
```

Comment:
is this type here: `Record<string, Record<number, HotspotDetails>>` a cache record? is it the same as `CacheRecord<HotspotDetails>`? seems like it could be a more generic/reusable type

Reply:
No, the `CacheRecord` only adds loading and a lastFetchedTimestamp to the data. So any object can become a cache record and get those extra properties to manage its caching lifecycle. Maybe its more clear now that I did the refactoring?